### PR TITLE
update parent, add missing osg headers

### PR DESCRIPTION
**Pull request by lukasj**
* update parent to 1.0.5
* add missing osgi headers
* avoid warnings during build

'mvn clean install' as well as 'mvn clean install -Poss-release' produces what I would consider as correct output. Double-check I used correct spec version etc numbers, please.

Thanks!
 
Fixes: #216

Signed-off-by: Lukas Jungmann <lukas.jungmann@oracle.com>

### CODE REVIEW

**Comment by lukasj** [comment]
@arjantijms can you take a look, please? Thx!

File: pom.xml [hunk 1]
```diff
@@ -21,17 +21,82 @@
...
+              - update the project <version> to not have -bXX
+        -->
+        <spec.title>Java(TM) Message Service Specification</spec.title>
+        <spec.version>2.0</spec.version>
+        <spec.newVersion>2.1</spec.newVersion>
+        <spec.build>01</spec.build>
+        <spec.impl.version>2.0.2</spec.impl.version>
+        <spec.extension.name>javax.jms</spec.extension.name>
+        <spec.non.final>false</spec.non.final>
+        <spec.api.package>jakarta.jms</spec.api.package>
```
Comment (arjantijms) [inline pom.xml]: Should this not be javax.jms? For this release the package name did not change.

Comment (lukasj) [reply pom.xml]: It should but then it "generates" javax.jms bundle symbolic name and that leads to a mismatch between file name and symbolic name. For some reason osgi tools (p2, tycho,...) expects these to match. OSGi spec also recommends these to match (with exception of using '_' as a separator between name and version). And in https://wiki.eclipse.org/JakartaEE_Maven_Versioning_Rules you can see, that symbolic name and file name are supposed to match as well.

As soon as the specversion plugin appears in eclipse-ee4j, someone should update it to jakarta rules. As it is now, it is used to define proper version numbers

Comment (arjantijms) [reply pom.xml]: Okay, thanks for the explanation. @m0mus Do we need to do a final check for all artefacts if they confirm to this?

**Pull request merged**